apps/meteor/app/threads/client/messageAction/useUnFollowMessageAction.ts

```diff
@@ -0,0 +1,66 @@
+import type { IMessage, IRoom, IUser } from '@rocket.chat/core-typings';
+import { isOmnichannelRoom } from '@rocket.chat/core-typings';
+import { useSetting, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
+import { useEffect } from 'react';
+
+import { useToggleFollowingThreadMutation } from '../../../../client/views/room/contextualBar/Threads/hooks/useToggleFollowingThreadMutation';
+import { Messages } from '../../../models/client';
+import { MessageAction } from '../../../ui-utils/client';
+import type { MessageActionContext } from '../../../ui-utils/client/lib/MessageAction';
+import { t } from '../../../utils/lib/i18n';
+
+export const useUnFollowMessageAction = (
+	message: IMessage,
+	{ room, user, context }: { room: IRoom; user: IUser | undefined; context: MessageActionContext },
+) => {
+	const threadsEnabled = useSetting('Threads_enabled');
+
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const toggleFollowingMutation = useToggleFollowingThreadMutation({
+		onSuccess: () => {
+			dispatchToastMessage({
+				type: 'success',
+				message: t('You_unfollowed_this_message'),
+			});
+		},
+	});
+	useEffect(() => {
+		if (!message || !threadsEnabled || isOmnichannelRoom(room)) {
+			return;
+		}
+
+		const { tmid, _id } = message;
+		let { replies } = message;
+
+		if (tmid || context) {
+			const parentMessage = Messages.findOne({ _id: tmid || _id }, { fields: { replies: 1 } });
```

Comment:
⚠️ `Messages.findOne()` is reactive! Consider using `useReactiveQuery()` to get `parentMessage`.